braindecode/models/spdnet.py

```diff
@@ -0,0 +1,238 @@
+    of a symmetric matrix.
+
+    Parameters
+    ----------
+    threshold : float
+        Threshold for the rectified linear unit
+
+    Returns
+    -------
+    torch.Tensor
+        Modified symmetric matrix
+
+    References
+    ----------
+    .. [1] Zhiwu Huang and Luc Van Gool, 2016,
+           A Riemannian Network for SPD Matrix Learning
+           AAAI
+    """
+    def __init__(self, threshold=1e-4):
+        super(ReEig, self).__init__()
+        self.register_buffer("threshold_", torch.tensor(threshold))
+
+    def forward(self, X):
+        return regm.apply(X, self.threshold_)
+
+
+class LogEig(nn.Module):
+    """LogEig layer from [1]_.
+
+    This class perform Riemannian projection into a flat space
+    by applying the logarithm to the eigenvalues of a symmetric matrix.
+
+    Parameters
+    ----------
+    dim : int
+        Dimension of the symmetric matrix
+    tril : bool
+        If True, only the lower triangular part of the matrix is used
+
+    Returns
+    -------
+    torch.Tensor
+        Modified symmetric matrix
+
+    References
+    ----------
+    .. [1] Zhiwu Huang and Luc Van Gool, 2016,
+           A Riemannian Network for SPD Matrix Learning
+           AAAI
+    """
+
+    def __init__(self, dim, tril=True, ):
+        super(LogEig, self).__init__()
+        self.tril = tril
+        if self.tril:
+            idx_lower = torch.tril_indices(dim, dim, offset=-1)
+            idx_diag = torch.arange(start=0, end=dim, dtype=torch.long)
+            self.idx = torch.cat((idx_diag[None, :].tile((2, 1)), idx_lower), dim=1)
+        self.dim = dim
+
+    def forward(self, X):
+        return self.embed(logm.apply(X))
+
+    def embed(self, X):
+        if self.tril:
+            x_vec = X[:, self.idx[0], self.idx[1]]
+            x_vec[:, self.dim:] *= 2 ** 0.5
+        else:
+            x_vec = X.flatten(start_dim=1)
+        return x_vec
+
+
+class SPDNet(EEGModuleMixin, nn.Module):
+    """SPDNet from [1]_.
+
+    This class is a SPDNet implementation from [1]_.
+
+    Parameters
+    ----------
+    input_type : str
+        Input type
+        If "raw", the input is a batch of raw EEG signals and
+        a covariance matrix is computed
+        If "cov", the input is a batch of covariance matrices and
+        the covariance matrix is used as input
+    n_chans : int
+        Number of channels
+    subspacedim : int
+        Subspace dimension
+    threshold : float
+        Threshold for the rectified linear unit
+    n_outputs : int
+        Output shape
+    tril : bool
+        If True, only the lower triangular part of the matrix is used
+
+    References
+    ----------
+    .. [1] Zhiwu Huang and Luc Van Gool, 2016,
+           A Riemannian Network for SPD Matrix Learning
+           AAAI
+    """
+    def __init__(
+        self,
+        input_type,
+        n_chans,
+        subspacedim,
+        threshold=1e-4,
+        n_outputs=1,
+        chs_info=None,
```

Comment:
check that all parameters are documented in the docstring

Reply:
Only some of the parameters are documented, but it is the same in several models in braindecode. I didn't know if I needed to add them or not. If this is a mistake in other models, a fix doc PR is needed. WDYT @bruAristimunha ?

Reply:
does it mean that here chs_info cannot be set (/ does not make sense) for SPDNet?

Reply:
Pinging @PierreGtch